base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -352,6 +352,65 @@ where B: BlockchainBackend
             self.clear_all_reorgs()?;
         }
 
+        self.rebuild_payref_indexes_task()?;
+
+        Ok(())
+    }
+
+    fn rebuild_payref_indexes_task(&self) -> Result<(), ChainStorageError> {
+        let initial_status = {
+            let db = self.db_read_access()?;
+            db.fetch_payref_rebuild_status()?
+        };
+        if !initial_status.is_rebuilt {
+            let chain_metadata = {
+                let db = self.db_read_access()?;
+                db.fetch_chain_metadata()?
+            };
+            let db_rw_lock = self.db.clone();
```

Comment:
```suggestion
        if initial_status.is_rebuilt {
        return Ok(())
        }
            let chain_metadata = {
                let db = self.db_read_access()?;
                db.fetch_chain_metadata()?
            };
            let db_rw_lock = self.db.clone();
```
This will remove an entire indent in the code.